File: CS/WindowsApplication3/Custom Ribbon/MyRibbonPanelViewInfo.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.Drawing;
using DevExpress.Utils.Drawing;

namespace DXSample {
    public class MyRibbonPanelViewInfo : RibbonPanelViewInfo
    {
        public MyRibbonPanelViewInfo(RibbonViewInfo viewInfo) : base(viewInfo) { }

 

        protected override RibbonPageGroupViewInfo CreateGroupViewInfo(RibbonPageGroup group)
        {
            return new MyRibbonPageGroupViewInfo(ViewInfo, group);
        }

    }


    class MyRibbonPageGroupViewInfo : RibbonPageGroupViewInfo
    {
        public MyRibbonPageGroupViewInfo(RibbonViewInfo viewInfo, RibbonPageGroup group) : base(viewInfo, group)
        {
        }

        protected override RibbonPanelGroupPainter CreatePanelGroupPainter()
        {
            return new MyRibbonPanelGroupPainter();
        }
    }
}

File: VB/WindowsApplication3/Custom Ribbon/MyRibbonPanelViewInfo.vb
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports DevExpress.XtraBars.Ribbon.Drawing

Namespace DXSample

    Public Class MyRibbonPanelViewInfo
        Inherits RibbonPanelViewInfo

        Public Sub New(ByVal viewInfo As RibbonViewInfo)
            MyBase.New(viewInfo)
        End Sub

        Protected Overrides Function CreatePanelGroupPainter() As DevExpress.XtraBars.Ribbon.Drawing.RibbonPanelGroupPainter
            Return New MyRibbonPanelGroupPainter()
        End Function
    End Class
End Namespace
